application/helpers/frontend_helper.php

```diff
@@ -1897,20 +1897,30 @@ function doAssessment($surveyid, $returndataonly=false)
                 }
 
                 // If this is a question (and not a survey field, like ID), save asessment value
-                if (!empty($field['title']))
+                if ($field['qid'] > 0)
                 {
                     /**
                      * Allow Plugin to update assessment value
                      */
                     // Prepare Event Info
                     $event = new PluginEvent('afterSurveyQuestionAssessment');
                     $event->set('surveyId', $surveyid);
-                    $event->set('question', $field);
+                    $event->set('lang', $_SESSION['survey_'.$surveyid]['s_lang']);
```

Comment:
You're right : but old code are sometimes broken. 
PS remind : i'm not the one who decide here ;)